llvm-passes/ReadAttributes/ReadAttributes.cpp

```diff
@@ -0,0 +1,70 @@
+#include "llvm/Pass.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Constants.h"
+
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/Transforms/Utils/ModuleUtils.h"
+#include "llvm/Transforms/IPO/PassManagerBuilder.h"
+
+#ifndef NDEBUG
+#include "llvm/IR/Verifier.h"
+#include "llvm/Support/Debug.h"
+#endif
+
+#include <set>
+
+using namespace llvm;
+
+static const char jitAnnotation[] = "jitme";
+
+class ReadAttributes : public ModulePass {
+
+    public:
+
+        static char ID;
+        ReadAttributes() : ModulePass(ID) {}
+
+        bool runOnModule(Module &mod) override {
+            bool modified = false;
+            std::set<Function*> to_jit;
+
+            auto annotations = mod.getNamedGlobal("llvm.global.annotations");
+            if(!annotations) {
+                dbgs() << "No annotations found!\n";
+                return false;
+            }
+
+            auto annot_array = cast<ConstantArray>(annotations->getOperand(0));
+            for(auto const& op: annot_array->operands()) {
+                auto annot_struct = cast<ConstantStruct>(op);
+
+                if (auto fn = dyn_cast<Function>(annot_struct->getOperand(0)->getOperand(0))) {
+                    auto anno = cast<ConstantDataArray>(cast<GlobalVariable>(annot_struct->getOperand(1)->getOperand(0))->getOperand(0))->getAsCString();
+                    if(anno == jitAnnotation) {
+                        to_jit.insert(fn);
+                        fn->addFnAttr(anno);
+                        dbgs() << "Found 'jitme' annotation on function: " << fn->getName() << "\n";
+                    }
+                }
+            }
+
```

Comment:
that looks ok to me, *but* you should make this an analyse, not a regular pass. have a look to https://github.com/quarkslab/llvm-dev-meeting-tutorial-2015/blob/master/ReachableIntegerValues/ReachableIntegerValues.cpp for some guidelines on how to make an analyse